lib/memcached/memcached.rb

```diff
@@ -585,6 +595,15 @@ def stats(subcommand = nil)
     raise e
   end
 
+  # Causes get / cas operations to return NOT_FOUND instead of raising NotFound
+  def without_exceptions(&block)
+    prev = @raise_on_errors
+    @raise_on_errors = false
+    yield
+  ensure
+    @raise_on_errors = prev
+  end
```

Comment:
Is this really needed? I'm assuming this feature will be used by `memcached_store`, so making the whole store compatible doesn't seem too hard.

Reply:
I wanted a way to enable this behaviour selectively before committing to using it everywhere.

Reply:
Yeah, I get that, but we might as well do the full sweep.

Reply:
@casperisfine are you suggesting to completely remove the exception logic instead of adding the new method? I think so, but just to be sure

Reply:
Not remove it entirely, but just to have a `MemcachedStore` that always check the return value and never rescue NOT_FOUND.

I don't think it's much more work to handle all the codepaths so might as well.